optuna/storages/_in_memory.py

```diff
@@ -63,7 +65,9 @@ def create_new_study(self, study_name: Optional[str] = None) -> int:
             else:
                 study_uuid = str(uuid.uuid4())
                 study_name = DEFAULT_STUDY_NAME_PREFIX + study_uuid
-            self._studies[study_id] = _StudyInfo(study_name)
+            study = _StudyInfo(study_name, list(directions))
+
+            self._studies[study_id] = study
```

Comment:
[nit] You can write it like:
```suggestion
            self._studies[study_id] = _StudyInfo(study_name, list(directions))
```